ccm_web/server/src/api/api.admin.handler.ts

```diff
@@ -84,18 +84,22 @@ export class AdminApiHandler {
     }
     if (instructor !== undefined) queryParams.by_teachers = ['sis_login_id:' + instructor]
     if (courseName !== undefined) queryParams.search_term = courseName
-    const coursesApiPromises = accountIds.map(async (a) => await this.getAccountCourses(a, queryParams))
+    const coursesApiPromises = createLimitedPromises<CanvasCourse[] | APIErrorData>(
+      accountIds.map(a => async () => await this.getAccountCourses(a, queryParams))
+    )
     const coursesResponses = await Promise.all(coursesApiPromises)
     const result = makeResponse<CanvasCourse[]>(coursesResponses)
     if (isAPIErrorData(result)) return result
     const allCourses: CanvasCourse[] = []
     result.map(cs => allCourses.push(...cs))
 
     // Get sections for those courses
-    const coursesWithSectionsApiPromises = allCourses.map(async c => {
-      const courseHandler = new CourseApiHandler(this.requestor, c.id)
-      return await courseHandler.getSectionsWithCourse(CourseApiHandler.slimCourse(c))
-    })
+    const coursesWithSectionsApiPromises = createLimitedPromises(
+      allCourses.map(c => async () => {
+        const courseHandler = new CourseApiHandler(this.requestor, c.id)
+        return await courseHandler.getSectionsWithCourse(CourseApiHandler.slimCourse(c))
```

Comment:
Also good idea to add to get sections call, I am not anticipating if a course will ever have more than 100 sections, but it is good to add.

Reply:
Similar to the comment about `getAccountCourses`, here we're limiting promises of calls to `courseHandler.getSectionsWithCourse`, so one promise per course. @kth/canvas-api handles the paging for getting sections. It still makes sense to me here because if there are lots of courses returned, we want to limit that.

Reply:
OK. I will go by your recommendation.